File: Lib/Telerik/Examples/QuickStart/GridView/CustomViews/ColumnChooserGridViewElement.vb
Imports System.Text
Imports Telerik.WinControls.UI
Imports Telerik.WinControls

Namespace Telerik.Examples.WinControls.GridView.CustomViews
	Public Class ColumnChooserGridViewElement
		Inherits GridVisualElement
		Implements IGridView
		#Region "Fields"

		Private titleBar As RadTitleBarElement
		Private columnChooser As ColumnChooserElement
		Private stackLayout As StackLayoutElement

		#End Region

		#Region "Initialization"

		Protected Overrides Sub InitializeFields()
			MyBase.InitializeFields()

			Me.StretchVertically = True
			Me.MinSize = New Size(200, 0)
		End Sub

		Protected Overrides Sub CreateChildElements()
			MyBase.CreateChildElements()

			Me.stackLayout = New StackLayoutElement()
			Me.stackLayout.Orientation = Orientation.Vertical
			Me.stackLayout.StretchVertically = True
			Me.stackLayout.StretchHorizontally = True
			Me.Children.Add(Me.stackLayout)

			Me.titleBar = New RadTitleBarElement()
			Me.titleBar.Text = "Column Chooser"
			Me.titleBar.StretchVertically = False
			Me.titleBar.AllowResize = False
			Me.titleBar.CloseButton.Visibility = ElementVisibility.Collapsed
			Me.titleBar.MinimizeButton.Visibility = ElementVisibility.Collapsed
			AddHandler titleBar.MaximizeRestore, AddressOf TitleBar_MaximizeRestore
			Me.stackLayout.Children.Add(Me.titleBar)

			Me.columnChooser = New ColumnChooserElement()
			Me.columnChooser.StretchVertically = True
			Me.stackLayout.Children.Add(Me.columnChooser)
        End Sub

        Private Sub SetElementStyles()
            Me.DrawFill = True
            Me.BackColor = Color.Gray
            Me.GradientStyle = GradientStyles.Solid
            Me.titleBar.BackColor = Color.WhiteSmoke
            Me.titleBar.ForeColor = Color.Black
        End Sub

		#End Region

		#Region "Event Handlers"

		Private Sub TitleBar_MaximizeRestore(ByVal sender As Object, ByVal args As EventArgs)
			Me.Visibility = ElementVisibility.Collapsed
			Me.GridViewElement.GridControl.ShowColumnChooser(Me.columnChooser.ViewTemplate)

			RemoveHandler GridViewElement.ColumnChooser.FormClosed, AddressOf ColumnChooser_FormClosed
			AddHandler GridViewElement.ColumnChooser.FormClosed, AddressOf ColumnChooser_FormClosed
		End Sub

		Private Sub ColumnChooser_FormClosed(ByVal sender As Object, ByVal e As FormClosedEventArgs)
			Me.Visibility = ElementVisibility.Visible
		End Sub

		#End Region

		#Region "IGridView Members"

		Public Sub Initialize(ByVal gridElement As RadGridViewElement, ByVal viewInfo As GridViewInfo) Implements IGridView.Initialize
            columnChooser.Initialize(gridElement, viewInfo)
            Me.SetElementStyles()
		End Sub

		Public Sub Detach() Implements IGridView.Detach
			columnChooser.Detach()
		End Sub

		Public Sub UpdateView() Implements IGridView.UpdateView
			columnChooser.UpdateView()
		End Sub

		Public ReadOnly Property GridViewElement() As RadGridViewElement Implements IGridView.GridViewElement
			Get
				Return columnChooser.GridViewElement
			End Get
		End Property

		Public ReadOnly Property ViewInfo() As GridViewInfo Implements IGridView.ViewInfo
			Get
				Return columnChooser.ViewInfo
			End Get
		End Property

		#End Region
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/GridView/CustomViews/ColumnChooserGridViewElement.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;
using Telerik.WinControls;
using System.Drawing;

namespace Telerik.Examples.WinControls.GridView.CustomViews
{
    public class ColumnChooserGridViewElement : GridVisualElement, IGridView
    {
        #region Fields

        private RadTitleBarElement titleBar;
        private ColumnChooserElement columnChooser;
        private StackLayoutElement stackLayout;

        #endregion

        #region Initialization

        protected override void InitializeFields()
        {
            base.InitializeFields();

            this.StretchVertically = true;
            this.MinSize = new Size(200, 0);
        }

        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            this.stackLayout = new StackLayoutElement();
            this.stackLayout.Orientation = System.Windows.Forms.Orientation.Vertical;
            this.stackLayout.StretchVertically = true;
            this.stackLayout.StretchHorizontally = true;
            this.Children.Add(this.stackLayout);

            this.titleBar = new RadTitleBarElement();
            this.titleBar.Text = "Column Chooser";
            this.titleBar.StretchVertically = false;
            this.titleBar.AllowResize = false;
            this.titleBar.CloseButton.Visibility = ElementVisibility.Collapsed;
            this.titleBar.MinimizeButton.Visibility = ElementVisibility.Collapsed;
            this.titleBar.MaximizeRestore += new TitleBarSystemEventHandler(TitleBar_MaximizeRestore);
            this.stackLayout.Children.Add(this.titleBar);

            this.columnChooser = new ColumnChooserElement();
            this.columnChooser.StretchVertically = true;
            this.stackLayout.Children.Add(this.columnChooser);
        }

        private void SetElementStyles()
        {
            this.DrawFill = true;
            this.BackColor = Color.Gray;
            this.GradientStyle = GradientStyles.Solid;

            this.titleBar.BackColor = Color.WhiteSmoke;
            this.titleBar.ForeColor = Color.Black;
        }

        #endregion

        #region Event Handlers

        private void TitleBar_MaximizeRestore(object sender, EventArgs args)
        {
            this.Visibility = ElementVisibility.Collapsed;
            this.GridViewElement.GridControl.ShowColumnChooser(this.columnChooser.ViewTemplate);

            this.GridViewElement.ColumnChooser.FormClosed -= new System.Windows.Forms.FormClosedEventHandler(ColumnChooser_FormClosed);
            this.GridViewElement.ColumnChooser.FormClosed += new System.Windows.Forms.FormClosedEventHandler(ColumnChooser_FormClosed);
        }

        private void ColumnChooser_FormClosed(object sender, System.Windows.Forms.FormClosedEventArgs e)
        {
            this.Visibility = ElementVisibility.Visible;
        }

        #endregion

        #region IGridView Members

        public void Initialize(RadGridViewElement gridElement, GridViewInfo viewInfo)
        {
            columnChooser.Initialize(gridElement, viewInfo);
            this.SetElementStyles();
        }

        public void Detach()
        {
            columnChooser.Detach();
        }

        public void UpdateView()
        {
            columnChooser.UpdateView();
        }

        public RadGridViewElement GridViewElement
        {
            get { return columnChooser.GridViewElement; }
        }

        public GridViewInfo ViewInfo
        {
            get { return columnChooser.ViewInfo; }
        }

        #endregion
    }
}
